### PR TITLE
Added text input component

### PR DESCRIPTION
**Pull request by tonywu315**

### CODE REVIEW

File: constellations/gemini-admin/src/utils/checkmark_green.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" ?><!DOCTYPE svg  PUBLIC '-//W3C//DTD SVG 1.1//EN'  'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'><svg height="512px" id="Layer_1" style="enable-background:new 0 0 512 512;" version="1.1" viewBox="0 0 512 512" width="512px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><path d="M448,71.9c-17.3-13.4-41.5-9.3-54.1,9.1L214,344.2l-99.1-107.3c-14.6-16.6-39.1-17.4-54.7-1.8  c-15.6,15.5-16.4,41.6-1.7,58.1c0,0,120.4,133.6,137.7,147c17.3,13.4,41.5,9.3,54.1-9.1l206.3-301.7  C469.2,110.9,465.3,85.2,448,71.9z" fill="#056571"/></svg>
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/utils/checkmark_green.svg:1]: place the images in a `gemini-admin/assets` folder

File: constellations/gemini-admin/webpack.common.js [hunk 2]
```diff
@@ -27,6 +27,17 @@ module.exports = {
...
                 loader: "style-loader!css-loader!sass-loader",
             },
+            {
+                test: /\.(png|jp(e*)g|svg|gif)$/,
+                use: [
+                    {
+                        loader: "file-loader",
+                        options: {
+                            name: "images/[hash]-[name].[ext]",
+                        },
```
Comment (ahsu1230) [inline constellations/gemini-admin/webpack.common.js:37]: I don't think you need this.

File: constellations/gemini-admin/src/help/help.sass [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
-    color: $colorGrayDark
-    
-  h2
-    font-family: $fontFamilyRoboto
-    font-size: $fontSize4
-    margin-left: 100px
-    padding-bottom: 20px
-    border-bottom: 1px solid
-    width: 1000px
+#view-help
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/help/help.sass:3]: ah sorry this is my bad, I missed this. @DanielLiu14

File: constellations/gemini-admin/src/utils/textInput.sass [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            border: 1px solid $colorGrayLight
+            line-height: $lineHeight0
+            font-size: $fontSize0
+
+        .blue
+            border-color: $colorLightBlue
+            border-width: 2px
+
+        img
+            width: $spacing4x
```
[the remark this answers is not in the document]
Comment (ahsu1230) [reply constellations/gemini-admin/src/utils/textInput.sass:36]: I created the $spacing__ variables mostly for a consistent unit in padding and margins. For images, it's fine to hard-code values.

File: constellations/gemini-admin/src/utils/textInput.sass [hunk 5]
```diff
@@ -0,0 +1,43 @@
+@import "../_.constants.sass"
+
+#text-input-wrapper
+    padding: $spacing1x 0
+
+    h2
+        font-size: $fontSize2
+
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/utils/textInput.sass:8]: I'd like to also differentiate the description from the label.
So can you give the description a color of `colorGrayMed`?

File: constellations/gemini-admin/src/utils/textInput.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+export class TextInput extends React.Component {
+    render = () => {
+        const description = this.props.description;
+        const required = this.props.required;
+        const value = this.props.value;
+
+        var format = <h4 className="hidden"></h4>;
+        if (!!description) {
+            let ending = required ? (
+                <span className="red">{" (required)"}</span>
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/utils/textInput.js:15]: One more functionality of this `required` part. If the input is satisfied (error message goes away and checkmark appears), turn this red into the normal font color.
The idea is that if the user did everything correctly, they should see no red on the page.

File: constellations/gemini-admin/src/programs/programEdit.js [hunk 7]
```diff
@@ -161,30 +147,98 @@ export class ProgramEditPage extends React.Component {
...
+                    description="Enter the higher grade"
+                    validators={[
+                        {
+                            validate: (grade2) => grade2 != "",
+                            message: "You must input a grade",
+                        },
+                        {
+                            validate: (grade2) => parseInt(grade2) <= 12,
+                            message: "Grade cannot be greater than 12",
+                        },
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/programs/programEdit.js:223]: We don't have to get too crazy with these validators since we do a lot of checking in the backend. But, we can add another easy one here.

```
{
         validate: (grade2) => this.state.inputGrade1 < parseInt(grade2),
         message: "Grade2 must be greater than Grade1",
},
```

File: constellations/gemini-admin/src/programs/programEdit.js [hunk 8]
```diff
@@ -1,43 +1,26 @@
...
+        oldProgramId: "",
+        inputProgramId: "",
+        inputProgramName: "",
+        inputGrade1: 0,
+        inputGrade2: 0,
+        inputDescription: "",
+    };
 
-    componentDidMount() {
+    componentDidMount = () => {
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/programs/programEdit.js:23]: thanks for fixing all of these

**Comment by ahsu1230** [comment]
Sorry one more comment. Let's rename this to `InputText`. Soon, we'll have other components that work similarly called `InputSwitch` and `InputCheckboxes` for example.

File: constellations/gemini-admin/src/help/help.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ import React from "react";
 export class HelpPage extends React.Component {
     render() {
         return (
-            <div>
+            <div id="view-help">
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/help/help.js:8]: Let's try to minimize the merge conflict @DanielLiu14 is going to have in #178 so let's get rid of all the changes in help-related stuff.

File: constellations/gemini-admin/src/utils/inputText.js [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+        var format = <h4 className="hidden"></h4>;
+        if (!!description) {
+            let ending;
+            if (required) {
+                if (pass) {
+                    ending = <span>{" (required)"}</span>;
+                } else {
+                    ending = <span className="red">{" (required)"}</span>;
+                }
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/utils/inputText.js:34]: I think you might be able to shorten this to:
```
ending = (<span {pass ? "" : "className=\"red\""}>{" (required)"}</span>);
```

File: constellations/gemini-admin/src/utils/inputText.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+            } else {
+                ending = " (optional)";
+            }
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/utils/inputText.js:37]: what do you think of a render_____() function to modularize this logic away from the main render() function?

File: constellations/gemini-admin/src/utils/inputText.js [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+            this.props.validators.some((validator) => {
+                if (!validator.validate(value)) {
+                    error = <h4 className="red">{validator.message}</h4>;
+                    pass = false;
+                    return true;
+                }
+            });
+        } else {
+            pass = false;
+        }
```
Comment (ahsu1230) [inline constellations/gemini-admin/src/utils/inputText.js:24]: what do you think of a renderErrorMessage() function to modularize this logic away from the main render() function?

Comment (tonywu315) [reply constellations/gemini-admin/src/utils/inputText.js:24]: Wouldn't I have to return both `pass` and `error`?
